app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt

```diff
@@ -95,19 +117,160 @@ class LoginFragment : Fragment(), LoginView {
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         if (resultCode == Activity.RESULT_OK) {
-            if (requestCode == REQUEST_CODE_FOR_CAS) {
-                data?.apply {
+            when (requestCode) {
+                REQUEST_CODE_FOR_CAS -> data?.apply {
                     presenter.authenticateWithCas(getStringExtra(INTENT_CAS_TOKEN))
                 }
-            } else if (requestCode == REQUEST_CODE_FOR_OAUTH) {
-                data?.apply {
-                    presenter.authenticateWithOauth(
-                        getStringExtra(INTENT_OAUTH_CREDENTIAL_TOKEN),
-                        getStringExtra(INTENT_OAUTH_CREDENTIAL_SECRET)
-                    )
+                REQUEST_CODE_FOR_OAUTH -> {
+                    isOauthSuccessful = true
+                    data?.apply {
+                        presenter.authenticateWithOauth(
+                            getStringExtra(INTENT_OAUTH_CREDENTIAL_TOKEN),
+                            getStringExtra(INTENT_OAUTH_CREDENTIAL_SECRET)
+                        )
+                    }
+                }
+                MULTIPLE_CREDENTIALS_READ -> {
+                    val loginCredentials: Credential =
+                        data!!.getParcelableExtra(Credential.EXTRA_KEY)
+                    handleCredential(loginCredentials)
                 }
+                NO_CREDENTIALS_EXIST -> {
+                    //use the hints to autofill sign in forms to reduce the info to be filled
+                    val loginCredentials: Credential =
+                        data!!.getParcelableExtra(Credential.EXTRA_KEY)
+                    val email = loginCredentials.id
+                    val password = loginCredentials.password
+
+                    text_username_or_email.setText(email)
+                    text_password.setText(password)
+                }
+                SAVE_CREDENTIALS -> Toast.makeText(
+                    context,
+                    getString(R.string.message_credentials_saved_successfully),
+                    Toast.LENGTH_SHORT
+                ).show()
             }
+        } else if (requestCode == SAVE_CREDENTIALS) {
```

Comment:
Are you needing this `else if` here?  Looks like you used this code for testing purpose, so you can remove it since it does not show anything to the user.